src/npctalk.cpp

```diff
@@ -178,55 +178,145 @@ struct sub_effect_parser {
 };
 
 struct item_search_data {
-    itype_id id;
-    item_category_id category;
-    material_id material;
-    int calories = 0;
-    std::vector<flag_id> flags;
-    std::vector<flag_id> excluded_flags;
+
+    std::vector<str_or_var> id;
+    str_or_var category;
+    std::vector<str_or_var> material;
+    dbl_or_var calories;
+    std::vector<str_or_var> flags;
+    std::vector<str_or_var> excluded_flags;
     bool worn_only;
     bool wielded_only;
+    bool helded_only;
```

Comment:
I think you wanted `held_only`. Items can be worn, held or wielded.

Reply:
Is this an error "carried forward" from somewhere else in the code base?

Reply:
No it's my own mistake xd